stylelint/README.md

```diff
@@ -0,0 +1,91 @@
+## Allowed Custom Properties plugin
+
+The purpose of this plugin is to ensure that we're following our established conventions for Polaris custom properties, and only using custom properties that are generated Polaris tokens.
+
+### Enforced Rules
+
+- If a `--p-*` custom property is used it must match an existing Polaris [token](../../../../src/tokens/tokens.ts).
+- Local component custom properties must have a `--pc-` prefix.
+
+## How to use
+
+### Options:
+
+```ts
+interface PrimaryOptions {
+  /**
+   * A list of regular expressions or string literals that match custom properties
+   * that are allowed to be used.
+   */
+  allowedProperties: (string | RegExp)[];
+  /**
+   * A map of properties and their allowed custom properties represented as a list
+   * of regular expressions or string literals.
+   */
+  allowedValues: {[property: string]: (string | RegExp)[]};
+}
+```
+
+### How to configure:
+
+```js
+const stylelintConfig = {
+  rules: {
+    '@shopify/custom-properties-allowed-list': {
+      allowedProperties: ['/--pc-.+/'],
+      allowedValues: {
+        width: ['--p-space-0', '--p-space-1' /* etc... */],
```

Comment:
We don't use `width` so maybe we should remove this example? How would most people use this?